lib/exhal/navigation.ex

```diff
@@ -184,6 +180,24 @@ defmodule ExHal.Navigation do
     end
   end
 
+  defp _follow_links(client, [link], tmpl_vars, opts) do
+    [_follow_link(client, link, tmpl_vars, opts)]
+  end
+
+  defp _follow_links(client, links, tmpl_vars, opts) do
+    {:ok, tsup} = Task.Supervisor.start_link()
```

Comment:
Maybe this is a good argument for having exhal become an application?

Reply:
Let's see how it works before we push it to something with long running processes/supervisors. It's nice that this is just a functional library at the moment.